### PR TITLE
CON-2833-Markdown-Distinct-Substring-Lenght-Exercise

### PR DESCRIPTION
**Pull request by zanninso**
[CON-2833](https://01talent.atlassian.net/browse/CON-2833)

[CON-2833]: https://01talent.atlassian.net/browse/CON-2833?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: subjects/java/checkpoints/distinct-substring-lenght/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Distinct Substring length
+
+### Instructions
+
+Create a class `DistinctSubstringLength` that provides a method to find the length of the longest substring without repeating characters in a given string.
```
Comment (nprimo) [inline subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: What if a string is made of all unique char? Is the length of the "longest substring without repeating characters" supposed to be the length of the string?

Comment (nprimo) [inline subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: I would specify if this is case-sensitive or not. In other exercises, the checks where case insensitive while here it seems to be case sensitive (from the tests)

Comment (nprimo) [inline subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: I would add a definition of "substring": I only understood this from one of the tests.

For example:
>... a substring is a contiguous sequence of [characters](https://en.wikipedia.org/wiki/Character_(computing)) within a [string](https://en.wikipedia.org/wiki/String_(computer_science)).

From [Wikipedia](https://en.wikipedia.org/wiki/Substring)

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/distinct-substring-lenght/README.md (hunk 1 above)
Comment (zanninso) [reply subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: yes~

Comment (zanninso) [reply subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: i do it in purpose, not all the informations should be clearly defined in the subject the have to get the informations from the example too.

Comment (zanninso) [reply subjects/java/checkpoints/distinct-substring-lenght/README.md]: the default is to be case-sensitive, but there is no harm to specify that :)

File: subjects/java/checkpoints/distinct-substring-lenght/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ### Instructions
 
-Create a class `DistinctSubstringLength` that provides a method to find the length of the longest substring without repeating characters in a given string.
+Create a class `DistinctSubstringLength` that provides a method to find the length of the longest substring without repeating characters in case sensitive withen the given string.
```
Comment (nprimo) [inline subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: I would keep the first version of the sentence and add a short additional sentence (or an example) to show that the distinctions between characters should be case-sensitive.

An example of a sentence to be added:
```
The distinction between characters should be case-sensitive.
```

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/distinct-substring-lenght/README.md (hunk 2 above)
Comment (zanninso) [reply subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: done

[the remark this answers is not in the document]
Comment (zanninso) [reply subjects/java/checkpoints/distinct-substring-lenght/README.md:5]: done

**Pull request merged**